mp2-v1/src/values_extraction/mod.rs

```diff
@@ -146,22 +248,22 @@ pub fn compute_leaf_single_values_digest(id: u64, value: &[u8]) -> Digest {
 
     let packed_value = left_pad32(value).pack(Endianness::Big).to_fields();
 
-    let inputs: Vec<_> = iter::once(GFp::from_canonical_u64(id))
+    let inputs: Vec<_> = once(F::from_canonical_u64(id))
         .chain(packed_value)
         .collect();
     map_to_curve_point(&inputs)
 }
 
 /// Calculate `metadata_digest = D(id || slot)` for single variable leaf.
 pub fn compute_leaf_single_metadata_digest(id: u64, slot: u8) -> Digest {
-    map_to_curve_point(&[GFp::from_canonical_u64(id), GFp::from_canonical_u8(slot)])
+    map_to_curve_point(&[F::from_canonical_u64(id), F::from_canonical_u8(slot)])
 }
 
 /// Calculate `metadata_digest = D(key_id || value_id || slot)` for mapping variable leaf.
 pub fn compute_leaf_mapping_metadata_digest(key_id: u64, value_id: u64, slot: u8) -> Digest {
     map_to_curve_point(&[
-        GFp::from_canonical_u64(key_id),
-        GFp::from_canonical_u64(value_id),
-        GFp::from_canonical_u8(slot),
+        F::from_canonical_u64(key_id),
+        F::from_canonical_u64(value_id),
+        F::from_canonical_u8(slot),
     ])
 }
```

Comment:
All these digests computation method should be updated to the new specs. They could maybe use the digest methods you introduced in `ColumnGadget` and `MetadataGagdet`?